### PR TITLE
feat(export): Can get a zip of a dataset using `get --format zip`

### PR DESCRIPTION
**Pull request by dustmop**
As part of https://github.com/qri-io/qri/issues/1176, a user can get a zip of their dataset like this:

```
qri get me/my_ds --format zip
```

By default, this will create a zip in the current directory named `my_ds.zip`. Also add an `-o` flag to get, so a user could instead do:

```
qri get me/my_ds --format zip -o export.zip
```

If the command is running through a pipe or redirect, and no `-o` flag is given, instead write the zip's bytes to stdout, so this command works:

```
qri get me/my_ds --format zip > export.zip
```

The same thing happens on the API, where a request like `/me/my_ds?format=zip` will receive an application/zip with a body response of the zip's bytes.

Some refactorings:
* The zip file contains a "qri-ref.txt" file, which is created the same way as FSI's ".qri-ref" file.
* Move hidden_file to it's own package to support this.
* Expand on API's TestRunner a bit so it's usable for the zip_test.

### CODE REVIEW

File: base/hiddenfile/hidden_file.go [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // +build !windows
 
-package base
+package hiddenfile
```
Comment (b5) [inline base/hiddenfile/hidden_file.go:3]: how would you feel about merging both `linkfile` and this new `hiddenfile` into `fsi` instead? I feel like they're both part of our story around filesystem integration. Ended up needing to write a code spike to confirm no import cycles, here's a commit if you'd like to cherry-pick:

https://github.com/qri-io/qri/tree/hidden_and_link_file_to_fsi

File: base/archive/zip_test.go [hunk 2]
```diff
@@ -101,34 +112,36 @@ func TestWriteZipFullDataset(t *testing.T) {
 	}
 }
 
-func TestUnzipDatasetBytes(t *testing.T) {
-	path := zipTestdataFile("exported.zip")
-	zipBytes, err := ioutil.ReadFile(path)
-	if err != nil {
-		t.Fatal(err)
-	}
+// TODO(dustmop): Rewrite zip importing
```
Comment (b5) [inline base/archive/zip_test.go:115]: Can we get an issue for this?

File: api/zip_test.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/qri-io/dataset"
+)
+
+func TestGetZip(t *testing.T) {
```
Comment (b5) [inline api/zip_test.go:11]: Would prefer to see all of this moved into `get_test.go`

File: base/archive/zip.go [hunk 4]
```diff
@@ -5,124 +5,91 @@ import (
...
 
-		if ds.Viz.RenderedPath != "" {
-			if err = maybeWriteRenderedViz(ctx, store, zw, ds.Viz.RenderedPath); err != nil {
-				if errors.Is(err, context.DeadlineExceeded) {
-					err = nil
-				} else {
-					return err
-				}
-			}
+		// TODO(dustmop): The transform component outputs a json file, with a path string
```
Comment (b5) [inline base/archive/zip.go:64]: Would love to see an issue for this as well

File: api/zip_test.go (hunk 3 above)
Comment (dustmop) [reply api/zip_test.go:11]: Do you mean rename this to `get_test.go`? There isn't a file with that name in the `api/` package.

Comment (b5) [reply api/zip_test.go:11]: Ooooops, yes rename then!

**Pull request merged**